extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/MSQMetricUtils.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.msq.exec;
+
+import org.apache.druid.indexing.common.task.Task;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.msq.dart.controller.sql.DartSqlEngine;
+import org.apache.druid.msq.input.InputSpec;
+import org.apache.druid.msq.input.table.TableInputSpec;
+import org.apache.druid.msq.kernel.StageDefinition;
+import org.apache.druid.msq.sql.MSQTaskSqlEngine;
+import org.apache.druid.query.BaseQuery;
+import org.apache.druid.query.DruidMetrics;
+import org.apache.druid.query.QueryContext;
+import org.joda.time.Interval;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Utility methods for setting up dimensions on metrics.
+ */
+public class MSQMetricUtils
+{
+  /**
+   * Sets dimensions for Dart queries, as well as some common dimensions. Sets the following dimensions:
+   * <ul>
+   * <li>queryId</li>
+   * <li>sqlQueryId</li>
+   * <li>type</li>
+   * <li>engine</li>
+   * </ul>
+   */
+  public static void setDartDimensions(
```

Comment:
nit: these `setX` methods could live inside some `MSQMetriceEventBuilder` class which should be a sublass of `ServiceMetricEvent.Builder`